apps/web/server/routers/viewer.tsx

```diff
@@ -707,12 +709,30 @@ const loggedInViewerRouter = createProtectedRouter()
         data.avatar = await resizeBase64Image(input.avatar);
       }
 
-      await prisma.user.update({
+      const updatedUser = await prisma.user.update({
         where: {
           id: user.id,
         },
         data,
+        select: {
+          id: true,
+          username: true,
+          email: true,
+          metadata: true,
+        },
       });
+
+      // Notify stripe about the change
+      if (updatedUser && updatedUser.metadata && hasKeyInMetadata(updatedUser, "stripeCustomerId")) {
```

Comment:
@zomars Should I add the same flag to know if we are on self hosted? And don't execute any stripe code.

Reply:
Some self-hosted users will use Stripe Connect, If this is only for premium usernames then we can skip it

Reply:
Let me get this right, connect can be for self hosted users, but this it's only involves our user plan handling so I should add flag here for `IS_SELF_HOSTED`.

Reply:
Nevermind seems like is good to go 👍🏽